paddle/fluid/imperative/dygraph_grad_maker.h

```diff
@@ -226,6 +225,9 @@ class TracedGradOp {
     }
 
     auto var_wrappers = ToVarWrapperList<kRole>(vars);
+
+    var_wrappers = CopyVarWrapperList(var_wrappers);
```

Comment:
Why do we need to snapshot here? By the way, I say you are doing pointer operation in CopyVarWrapperList, so it is not "Copy" VarWrapperList

Reply:
> Why do we need to snapshot here? 

Because the current `inplace_version` of `Variable` should be saved here, and it will be used to check whether there is any inplace operation affecting gradient calculation.

> By the way, I say you are doing pointer operation in CopyVarWrapperList, so it is not "Copy" VarWrapperList

Thanks, "Copy" will be changed